common/ts/helpers/request.ts

```diff
@@ -26,18 +27,59 @@ export function getJSON(endpoint: Endpoint, path: string, query?: RequestData):
       },
       (error, response, body) => {
         if (error) {
-          const errMsg = `[SQ] API GET '${path}' failed, error was: ${JSON.stringify(error)}`;
-          tl.debug(errMsg);
-          return reject(new Error(errMsg));
+          return logAndReject(
+            reject,
+            `[SQ] API GET '${path}' failed, error was: ${JSON.stringify(error)}`
+          );
         }
         tl.debug(`Response: ${response.statusCode} Body: "${JSON.stringify(body)}"`);
         if (response.statusCode < 200 || response.statusCode >= 300) {
-          const errMsg = `[SQ] API GET '${path}' failed, status code was: ${response.statusCode}`;
-          tl.debug(errMsg);
-          return reject(new Error(errMsg));
+          return logAndReject(
+            reject,
+            `[SQ] API GET '${path}' failed, status code was: ${response.statusCode}`
+          );
         }
         return resolve(body || {});
       }
     );
   });
 }
+
+export function getServerVersion(endpoint: Endpoint): Promise<semver.SemVer> {
```

Comment:
I think you are basically duplicating all the code from `getJSON` (expect for the `json: true` option). So maybe it would be better to create a `get` method with nearly the same signature as `getJSON`, it could also take some optional options (so that `getJSON`, could use it and pass `{json: true}` as an option) to prevent code duplication.

And then you could use it like this: 
```ts
get(endpoint, '/api/server/version').then(semver.coerce)
```